sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_directory_client.hpp

```diff
@@ -8,7 +8,7 @@
 #include <vector>
 
 #include <azure/core/credentials.hpp>
-#include <azure/core/http/pipeline.hpp>
+#include <azure/core/internal/http/pipeline.hpp>
```

Comment:
What can we do so storage public headers don't expose internal headers like this?

Reply:
We would need to move the pipeline creation to cpp and avoid using it as a parameter for functions.